plugin/src/main/java/org/jenkinsci/plugins/casc/ConfigurationAsCode.java

```diff
@@ -158,15 +156,23 @@ public void doReplace(StaplerRequest request, StaplerResponse response) throws E
             sources = Collections.singletonList(newSource);
             //TODO: here should be dry run
             configureWith(getConfigFromSources(getSources()));
-            lastMessage = "success";
+            LOGGER.log(Level.INFO, "Replace configuration with: " + file.getAbsolutePath());
         } else {
-            lastMessage = "No such file exists";
+            LOGGER.log(Level.INFO, "There is no such source exist, applying default");
             configure();
         }
-        LOGGER.log(Level.INFO, "Replace configuration");
         response.sendRedirect("");
     }
 
+    public FormValidation doCheckNewSource(@QueryParameter String newSource) {
+        LOGGER.log(Level.INFO, "Validation was called");
+        File file = new File(newSource);
```

Comment:
You should definitely check for null and use `Util.fixEmptyAndTrim` then return `FormValidation.ok()` otherwise it would be blocking

Reply:
@casz Did you check? Does it work for you? I can see from my logs the function is not even called.

Reply:
I'll have a look later today, once I get off work 😅